xdem/spatial_tools.py

```diff
@@ -396,3 +398,102 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # Return the hillshade, scaled to uint8 ranges.
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
+
+
+@numba.njit
+def _get_closest_rectangle(size: int) -> tuple[int, int]:
+    """
+    Given a 1D array size, return a rectangular shape that is closest to a cube which the size fits in.
+
+    If 'size' does not have an integer root, a rectangle is returned that is slightly larger than 'size'.
+    
+    :examples:
+        >>> _get_closest_rectangle(4)  # size will be 4
+        (2, 2)
+        >>> _get_closest_rectangle(9)  # size will be 9
+        (3, 3)
+        >>> _get_closest_rectangle(3)  # size will be 4; needs padding afterward.
+        (2, 2)
+        >>> _get_closest_rectangle(55) # size will be 56; needs padding afterward.
+        (8, 7)
+        >>> _get_closest_rectangle(24)  # size will be 25; needs padding afterward
+        (5, 5)
+        >>> _get_closest_rectangle(85620)  # size will be 85849; needs padding afterward
+        (293, 293)
+        >>> _get_closest_rectangle(52011)  # size will be 52212; needs padding afterward
+        (229, 228)
+    """
+    close_cube = int(np.sqrt(size))
+
+    if close_cube ** 2 == size:
+        return (close_cube, close_cube)
+
+    height = close_cube
+    width = close_cube
+    # Iteratively increase height, then width, until the size is larger or equal to the input size
+    for i in range(int(1e8)):
```

Comment:
I don't think this number of iterations is ever used...
First, you find N such that N^2 < size < (N+1)^2 (here N is your close_cube).
Then your final result is always either N^2, N*(N+1) or (N+1)^2, depending on which one is larger than size first.

Reply:
Good point! I avoid `while True` because I read somewhere that NASA avoids it haha (to avoid endless loops). That loop should never finish; it's just a precaution for potential flaws in my programming.

Reply:
Ok, but my point was also that the loop is probably not needed. You only need to test N, N*(N+1) and (N+1)^2.

Reply:
So you loop only twice instead of 1e8!

Reply:
Could you please elaborate on what you mean, e.g. though a minimal code example? I'm unfortunately not sure of what you mean!

Reply:
Let me state it again differently. If size is the total number of pixels in your "close to cube", then we can always find an integer, let's call it N for simplicity (it's called close_cube in your code) that verifies:
_N^2 <= size < (N+1)^2_.
You find N as you do with `int(np.sqrt(size))`. So the possibilities for your "close to cube" are a rectangle of dimensions _N x N_, _N x (N+1)_ or _(N+1) x (N+1)_. Just pick the one among these three that is larger than size (by construction, the 3rd one will always be large enough) and that's it. You do not need 1e8 iterations to find that out, just math! ;-)

Reply:
Implemented in 58a021b. I think the loop never iterated more than at most two times, so it's much better to remove it.

Reply:
Exactly, this is the first sentence I wrote! ;-)